OneSignalSDK/onesignal/src/main/java/com/onesignal/GcmJobIntentService.java

```diff
@@ -0,0 +1,32 @@
+package com.onesignal;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Parcelable;
+import android.os.PersistableBundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresApi;
+import android.support.v4.app.JobIntentService;
+
+@RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+public class GcmJobIntentService extends JobIntentService {
+
+    public static final String BUNDLE_EXTRA = "Bundle:Parcelable:Extras";
+    private static final int JOB_ID = 123890;
+
+    @Override
+    protected void onHandleWork(@NonNull Intent intent) {
+        Parcelable parcelable = intent.getExtras().getParcelable(BUNDLE_EXTRA);
+        BundleCompat bundle = parcelable instanceof PersistableBundle && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1?
```

Comment:
Small nit, I think we can clean this up a bit by using `BundleCompatFactory`. Feel free to add a new method to `BundleCompat` or  `BundleCompatFactory` if needed.
Or if it really doesn't help it is ok to leave this code as is.

Reply:
👍 Looks cleaner now! Thanks.